### DISCUSSION TITLE
Stream POST no response body

### DISCUSSION THREAD

**Discussion by ramtinsoltani** [sindresorhus/got#1626] [Q&A] [answered]
#### What would you like to discuss?

I'm using the stream API to upload a file:
```ts
const stream = fs.createReadStream(path.resolve(process.cwd(), 'test.txt'));
const req = got.stream('http://localhost:5000/file/test.txt', {
  headers: {
    'Content-Type': 'application/octet-stream'
  },
  method: 'post',
  throwHttpErrors: false,
  retry: 0,
  responseType: 'json'
})
.on('response', res => {

  console.log(res.body);

})
.on('error', error => {

  stream.close();
  console.error(error);

});

stream.pipe(req);
```

The `response` event is emitted but `res.body` and `res.rawBody` are always `undefined`. The local server responds with a JSON body when the file upload is complete.

How would I read the response body?

#### Checklist

- [x] I have read the documentation.

**Comment by szmarczak** [comment, marked as answer]
> How would I read the response body?

It is a stream, not a promise, so you shouldn't go for `response.body`. Use  [`get-stream`](https://github.com/sindresorhus/get-stream/).

**Reply by ramtinsoltani** [reply]
I know it's a stream. As I've mentioned in my question, this is a file upload. `get-stream` returns the content of the file I'm trying to upload.

I'm trying to read the `response.body` sent by the server after the file stream has ended. The `request` package used the `response` event to provide the final response after a file upload, which contained the parsed body sent by the server. However, this is not the case with `got`. As mentioned in the documentation, the stream API ignores `responseType` config, and the response object emitted on `response` won't have a `body` or a `rawBody` on the object.

I can't find a way to read the body sent by the server, and `get-stream` won't help with that.

**Reply by szmarczak** [reply]
> I'm trying to read the response.body sent by the server after the file stream has ended.
> I can't find a way to read the body sent by the server, and get-stream won't help with that.

It does. If it doesn't then you're doing something wrong.

https://github.com/sindresorhus/got/discussions/1626#discussioncomment-381630

**Reply by szmarczak** [reply]
Here's a test that proves it: https://github.com/sindresorhus/got/blob/7e95b4165f4d484bc309a1cdf222dc0917b7c8ea/test/stream.ts#L55-L60

**Reply by ramtinsoltani** [reply]
That worked! I was using `get-stream` on the file stream and not the request object returned by `got.stream()`.

Thanks for the help. It is much appreciated.
